xDiffCollectionTests/CollectionTestStatusTests.swift

```diff
@@ -0,0 +1,128 @@
+        XCTAssertEqual(myCollection!.element(atIndexPath: IndexPath(row: 1, section: 0)), object2)
+        XCTAssertEqual(myCollection!.element(atIndexPath: IndexPath(row: 2, section: 0)), object3)
+        XCTAssertEqual(myCollection!.element(atIndexPath: IndexPath(row: 3, section: 0)), object4)
+    }
+
+    func test_Update_Element_That_Matches_No_Filter_Should_Return_Three_Empty_Arrays() {
+        test_Fillup_Collection()
+        
+        var object = CollectionTestObjectMock(value: "NewObject")
+        object.status = .other
+        
+        let resp = myCollection!.update(element: object)
+        XCTAssert(resp.added.count == 0)
+        XCTAssert(resp.updated.count == 0)
+        XCTAssert(resp.deleted.count == 0)
+        
+        XCTAssertEqual(myCollection!.element(atIndexPath: IndexPath(row: 0, section: 0)), object1)
+        XCTAssertEqual(myCollection!.element(atIndexPath: IndexPath(row: 1, section: 0)), object2)
+        XCTAssertEqual(myCollection!.element(atIndexPath: IndexPath(row: 2, section: 0)), object3)
+        XCTAssertEqual(myCollection!.element(atIndexPath: IndexPath(row: 3, section: 0)), object4)
+    }
+    
+    func test_Update_Status_Should_Delete_From_And_Add_To_Bin() {
+        test_Fillup_Collection()
+        
+        var objectClone = object3
+        objectClone.status = .old
+        
+        var resp = myCollection!.update(element: objectClone)
+        XCTAssertEqual(resp.added, [IndexPath(row: 0, section: 1)])
+        XCTAssert(resp.updated.count == 0)
+        XCTAssertEqual(resp.deleted, [IndexPath(row: 2, section: 0)])
+
+        XCTAssertEqual(myCollection!.element(atIndexPath: IndexPath(row: 0, section: 1)), objectClone)
+        
+        
+        objectClone.status = .hot
+        resp = myCollection!.update(element: objectClone)
+        XCTAssertEqual(resp.added, [IndexPath(row: 0, section: 2)])
+        XCTAssert(resp.updated.count == 0)
+        XCTAssertEqual(resp.deleted, [IndexPath(row: 0, section: 1)])
+        
+        XCTAssertEqual(myCollection!.element(atIndexPath: IndexPath(row: 0, section: 2)), objectClone)
+    }
+
+    func test_Delete_Non_Existing_Item_Should_Return_Three_Empty_Arrays() {
+        test_Fillup_Collection()
+        
+        let object = CollectionTestObjectMock(value: "NewObject")
+        
+        let resp = myCollection!.delete(element: object)
+        XCTAssert(resp.added.count == 0)
+        XCTAssert(resp.updated.count == 0)
+        XCTAssert(resp.deleted.count == 0)
+        
+        XCTAssertEqual(myCollection!.element(atIndexPath: IndexPath(row: 0, section: 0)), object1)
+        XCTAssertEqual(myCollection!.element(atIndexPath: IndexPath(row: 1, section: 0)), object2)
+        XCTAssertEqual(myCollection!.element(atIndexPath: IndexPath(row: 2, section: 0)), object3)
+        XCTAssertEqual(myCollection!.element(atIndexPath: IndexPath(row: 3, section: 0)), object4)
+    }
```

Comment:
I'll put a general comment for the whole file.
Looks like all the tests have the same logic that could be abstracted for readability?
What do you think?